pages/docs/platform/monitor/open-telemetry.mdx

```diff
@@ -0,0 +1,196 @@
+  time to patch code in order to collect traces and spans from all parts of your
+  application. Loading running `otelMiddleware()` after any other code risks not
+  instrumenting it.
+</Callout>
+
+For example,
+
+```ts
+// Import this first
+import { otelMiddleware } from "inngest/experimental";
+const otel = otelMiddleware();
+
+// Then everything else
+import { Inngest } from "inngest";
+
+const inngest = new Inngest({
+  id: "my-app",
+  middleware: [otel],
+});
+```
+
+### Serverless
+
+If you're using serverless, the entrypoint of your app will likely be the file
+for a particular endpoint, for example `/api/inngest`.
+
+If you have your client set up as in the example above, make sure you import
+that first so that the provider has a chance to initialize.
+
+```ts
+// Import the client first
+import { inngest } from "@/inngest";
+
+// Then import everything else
+import { serve } from "inngest/next";
+import { myFn } from "@/inngest/functions";
+
+export const { GET, POST, PUT } = serve({
+  client: inngest,
+  functions: [myFn],
+});
+```
+
+### Extending existing providers
+
+A JavaScript process can only have a single OpenTelemetry Provider. Some
+libraries such as Sentry also create their own provider.
+
+`otelMiddleware()` will first try to _extend_ an existing provider and will only
+create one if none has been found. If an existing provider is extended, we won't
+contribute any automatic [instrumentation](#instrumentation).
+
+In the case of Sentry, `otelMiddleware()` will extend Sentry's provider as long
+as it's run after `Sentry.init()`.
+
+<Note>
+  This extension should also work for OpenTelemetry providers that originate
+  within the runtime, like [Deno's OpenTelemetry](https://docs.deno.com/runtime/fundamentals/open_telemetry/).
+</Note>
+
+This behaviour can be changed:
+
+```ts
+otelMiddleware({
+  behaviour: "auto",
+});
+```
+
+The options are:
+
+- `"auto"` (default): Attempt to extend a provider if one exists, else create one, fails
+  if neither worked
+- `"extendProvider"`: Only attempt to extend a provider and fails if none exists
+- `"createProvider"`: Only attempt to create a provider and fails if we couldn't
+- `"off"`: Do nothing
+
+If you're intending to only use `otelMiddleware()` to extend an existing
+provider, you no longer need to ensure that it is called before any other code.
+
+### Manually extend
+
+If you're already manually creating your own trace provider and import ordering
+is an issue, you may want to manually add Inngest's `InngestSpanProcessor` to
+your existing setup.
+
+Add an `InngestSpanExporter` to your provider:
+
+```ts
+// Create your client the same as you would normally
+import { Inngest } from "inngest";
+import { otelMiddleware } from "inngest/experimental";
+
+export const inngest = new Inngest({
+  id: "my-app",
+  middleware: [
+    otelMiddleware({
+      // Make sure the middleware doesn't try to automatically instrument
+      behaviour: "off",
+    }),
+  ],
+});
+```
```

Comment:
```suggestion
```ts
// Create your client the same as you would normally
import { Inngest } from "inngest";
import { extendedTracesMiddleware } from "inngest/experimental";

export const inngest = new Inngest({
  id: "my-app",
  middleware: [
    extendedTracesMiddleware({
      // Make sure the middleware doesn't try to automatically instrument
      behaviour: "off",
    }),
  ],
});
```
```